p2p-tests/implicitly-tracked-peer-branches-test.ts

```diff
@@ -0,0 +1,189 @@
+      },
+      description: "",
+      defaultBranch: "main",
+    });
+  });
+
+  // Assert that the seed received the project.
+  await withRetry(async () => {
+    const result = radCli({
+      radHome: seed.radHome,
+      args: ["identities", "project", "get", "--urn", project.urn],
+    });
+
+    strictAssert.deepStrictEqual(result, {
+      urn: project.urn,
+      payload: {
+        "https://radicle.xyz/link/identities/project/v1": {
+          name: "my-fancy-project",
+          description: "",
+          default_branch: "main",
+        },
+      },
+    });
+  });
+
+  // Without this the test fails, not sure why.
+  await sleep(3000);
+
+  // Contributor follows the project.
+  await withRetry(async () => {
+    await contributor.proxyClient.project.requestSubmit(project.urn);
+  });
+
+  // Assert that the contributor received the project.
+  await withRetry(async () => {
+    const result = await contributor.proxyClient.project.get(project.urn);
+
+    strictAssert.deepStrictEqual(result.urn, project.urn);
+  });
+
+  // Contributor forks the project.
+  await contributor.proxyClient.project.checkout(project.urn, {
+    path: contributor.checkoutPath,
+    peerId: maintainer.peerId,
+  });
+
+  // Assert that the seed received the contributor's fork.
+  await withRetry(async () => {
+    const result = radCli({
+      radHome: seed.radHome,
+      args: ["identities", "project", "tracked", "--urn", project.urn],
+    });
+
+    strictAssert.deepStrictEqual(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (result as any).find((x: any) => x.peerId === contributor.peerId).status
+        .type,
+      "replicated"
+    );
+  });
+
+  // Maintainer adds contributor as a remote to the project.
+  await withRetry(async () => {
+    await maintainer.proxyClient.project.peerTrack(
+      project.urn,
+      contributor.peerId
+    );
+  });
+
+  // Assert that the maintainer found the remote.
+  await withRetry(async () => {
+    const result = await maintainer.proxyClient.project.listPeers(project.urn);
+
+    strictAssert.deepStrictEqual(
+      result.find(x => x.peerId === contributor.peerId)?.status.type,
+      "replicated"
+    );
+  });
+
+  // Assert that the maintainer can view the contributor's branch.
+  await withRetry(async () => {
+    const branches = await maintainer.proxyClient.source.branchesGet({
+      projectUrn: project.urn,
+      peerId: contributor.peerId,
+    });
+
+    strictAssert.deepStrictEqual(branches, ["main"]);
+  });
+
+  // Contributor2 follows the project.
+  await withRetry(async () => {
+    await contributor2.proxyClient.project.requestSubmit(project.urn);
+  });
+
+  // Assert that contributor2 received the project.
+  await withRetry(async () => {
+    const result = await contributor2.proxyClient.project.get(project.urn);
+
+    strictAssert.deepStrictEqual(result.urn, project.urn);
+  });
+
+  // Assert that contributor2 can view contributor's branch.
```

Comment:
Can we add an assertion before this: That “contributor” is in contributor2’s peer list? This is the confusing part, right?

Reply:
Good point. Now that I added this check, I can't get this test to do the implicit tracking part. Do you have any idea what we might be missing here?

I tried adding a commit from `contributor`, but that didn't help either. `contributor2` only sees the `maintainer` and itself in the peer list.

Here's the check I added: 5796a02f7561973a57a6b057e89820221e1628d3.

```
maint:  hybhe78oy41yzux5d6rk8fdtek881de76wuwy7mga1imzoj7p17pnw
cont2:  hynggyf6eftw63a5emj77afqgji69e1i3gicgot61omx6tmbekgkxy
cont:  hydkjndj7zhq98wg15akseowbtykjpym15oc613jq8frbpoisgi4pn
[
  {
    type: 'local',
    peerId: 'hynggyf6eftw63a5emj77afqgji69e1i3gicgot61omx6tmbekgkxy',
    status: { type: 'replicated', role: 'tracker', user: [Object] }
  },
  {
    type: 'remote',
    peerId: 'hybhe78oy41yzux5d6rk8fdtek881de76wuwy7mga1imzoj7p17pnw',
    status: { type: 'replicated', role: 'maintainer', user: [Object] }
  }
]
```